storage-narpc/src/main/java/org/apache/crail/storage/tcp/TcpStorageServer.java

```diff
@@ -176,13 +176,19 @@ public static InetSocketAddress getDataNodeAddress() throws IOException {
 				InetAddress _addr = address.getAddress();
 				addr = _addr;
 			}
-		}		
+		}
+
+		if (addr == null){
+			throw new IOException("Cannot find valid network interface with name " + TcpStorageConstants.STORAGE_TCP_INTERFACE);
```

Comment:
The message is a bit vague. Why not: "Network interface with name XXX has no valid IP address"